external-crates/move/move-compiler/src/to_bytecode/translate.rs

```diff
@@ -663,8 +664,23 @@ fn function(
     )
 }
 
-fn visibility(v: Visibility) -> IR::FunctionVisibility {
+fn visibility(
+    context: &mut Context,
+    function_name: &FunctionName,
+    entry: Option<Loc>,
+    v: Visibility,
+) -> IR::FunctionVisibility {
     match v {
+        _ if is_public_for_testing(
+            context.env,
+            context.current_package(),
+            function_name,
+            entry,
+        )
+        .is_some() =>
+        {
+            IR::FunctionVisibility::Public
```

Comment:
question for my own knowledge: will this mark the on-chain function as public? if so, what prevents other packages from calling it?

Reply:
This is only for unit tests, so there is no "on-chain" here